metagraph-plugin/tests/test_algorithms.py

```diff
@@ -2,32 +2,30 @@
 import numpy as np
 import pytest
 
-
-# directed graph
-@pytest.fixture(autouse=True)
-def kg_from_nx_di_8_12(networkx_weighted_directed_8_12):
-    pg_test_case = mg.translate(networkx_weighted_directed_8_12, mg.wrappers.Graph.KatanaGraph)
-    return pg_test_case
-
-
-# undirected graph
-@pytest.fixture(autouse=True)
-def kg_from_nx_ud_8_12(networkx_weighted_undirected_8_12):
-    pg_test_case = mg.translate(networkx_weighted_undirected_8_12, mg.wrappers.Graph.KatanaGraph)
-    return pg_test_case
-
-
 def test_bfs(networkx_weighted_directed_8_12, kg_from_nx_di_8_12):
     bfs1_nx = mg.algos.traversal.bfs_iter(networkx_weighted_directed_8_12, 0)
     bfs2_nx = mg.algos.traversal.bfs_iter(networkx_weighted_directed_8_12, 2)
-    bfs1_kg = mg.algos.traversal.bfs_iter(kg_from_nx_di_8_12, 0, 2 ** 30 - 1)
-    bfs2_kg = mg.algos.traversal.bfs_iter(kg_from_nx_di_8_12, 2, 2 ** 30 - 1)
+    bfs1_kg = mg.algos.traversal.bfs_iter(kg_from_nx_di_8_12, 0)
+    bfs2_kg = mg.algos.traversal.bfs_iter(kg_from_nx_di_8_12, 2)
     assert bfs1_kg.tolist() == bfs1_nx.tolist()
     assert bfs2_kg.tolist() == bfs2_nx.tolist()
     assert bfs1_kg.tolist() == [0, 1, 3, 4, 7]
     assert bfs2_kg.tolist() == [2, 4, 5, 6, 7]
 
 
+# test for katana graph which is directly loaded rather than translated from nettworkx
+# also test two consecutive runs with the same source code
+def test_bfs_kg(katanagraph_rmat15_cleaned_di, nx_from_kg_rmat15_cleaned_di):
+    src_node = 10
+    bfs1_kg = mg.algos.traversal.bfs_iter(katanagraph_rmat15_cleaned_di, src_node)
+    bfs2_kg = mg.algos.traversal.bfs_iter(katanagraph_rmat15_cleaned_di, src_node)
+    bfs_nx = mg.algos.traversal.bfs_iter(nx_from_kg_rmat15_cleaned_di, src_node)
+    assert bfs1_kg.tolist() == bfs2_kg.tolist()
+    assert len(bfs1_kg.tolist()) > 0
+    # assert bfs1_kg.tolist() == bfs_nx.tolist() # TODO(pengfei): replace katanagraph_rmat15_cleaned_di with a cleaned version and uncomment this
```

Comment:
I'm a bit confused. We have cleaned versions of this graph I think. Are they not in the right format in the input data blob?

Reply:
A short answer is yes. "cleaned" here I think only means the duplicated edges are removed. But I can still see the nodes which has no edges (I tested this using the scripts below). This will cause an error (node index > # of nodes) when we are using networkx's built-in algorithms. 

Any suggestions about that?

```
import katana.local
from katana.example_data import get_input
from katana.local import Graph
katana.local.initialize()
pg = Graph(get_input("propertygraphs/rmat15_cleaned_symmetric"))
num_no_edge_nodes = 0
for nid in pg:
for nid in pg:
    if pg.edges(nid)==range(0,0):
        num_no_edge_nodes += 1
print (num_no_edge_nodes)
```

Reply:
@arthurp